funnel/views/proposal.py

```diff
@@ -386,21 +390,24 @@ def proposal_prev(profile, space, proposal):
         return redirect(space.url_for())
 
 
-@app.route('/<space>/<proposal>/move', subdomain='<profile>')
+@app.route('/<space>/<proposal>/move', methods=['POST',], subdomain='<profile>')
 @load_models(
     (Profile, {'name': 'profile'}, 'g.profile'),
     ((ProposalSpace, ProposalSpaceRedirect), {'name': 'space', 'profile': 'profile'}, 'space'),
     ((Proposal, ProposalRedirect), {'url_name': 'proposal', 'proposal_space': 'space'}, 'proposal'),
     permission='move-proposal', addlperms=lastuser.permissions)
 def proposal_moveto(profile, space, proposal):
-    target_name = request.args.get('target')
-    if not target_name:
-        abort(404)
-    profile_name, space_name = target_name.split('/', 1)
-    target_space = ProposalSpace.query.filter(ProposalSpace.name == space_name).join(Profile).filter(Profile.name == profile_name).one_or_none()
-    if target_space:
-        proposal.move_to(target_space)
-        db.session.commit()  # WARNING: GET request!
+    proposal_move_form = ProposalMoveForm()
+    if not proposal_move_form.validate_on_submit():
+        flash(_("Please choose a proposal space you want to move this proposal to."))
         return redirect(proposal.url_for(), 303)
-    else:
-        abort(404)
+
+    target_space = proposal_move_form.target.data
+    if not (target_space.profile.admin_team in g.user.teams or target_space.admin_team in g.user.teams):
```

Comment:
Is this really required? Doesn't the permission check for `move-proposal` on L398 suffice?

Reply:
@shreyas-satish does L398 permission check if the user has permission to move proposal to the target profile?

Reply:
I think the permission in L398 is assigned based on current user and current proposal space, right?

Reply:
> I think the permission in L398 is assigned based on current user and current proposal space, right?

@iambibhas Try it out?

Reply:
@shreyas-satish L398 says that you have `move-proposal` permission for current proposal space. We need to check explicitly if the user has same permission for the space we want to move to. Hence this check.